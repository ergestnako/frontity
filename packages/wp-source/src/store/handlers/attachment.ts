import { Handler } from "../types";
import { action } from "overmind";

const attachmentHandler: Handler = async (ctx, { name, params }) => {
  const state = ctx.state.source;
  const actions = ctx.actions.source;
  const effects = ctx.effects.source;

  const { slug } = params;

  // First, search attachment with the given slug
  let attachment: any = Object.values(state.post).find(
    (a: any) => a.slug === slug
  );

  // If none is found
  if (!attachment) {
    const { isOk, entities } = await effects.api.get({
      endpoint: "media",
      params: { slug, _embed: true }
    });

    attachment = entities[0];
    actions.populate({ entities });
  }

  // Init data
  if (attachment) {
    const { type, id, link } = attachment;
    Object.assign(state.data[name], {
      type,
      id,
      link,
      isPostType: true,
      isMedia: true,
      isAttachment: true,
    });
  } else {
    Object.assign(state.data[name], {
      is404: true
    });
  }
};

export default attachmentHandler;
