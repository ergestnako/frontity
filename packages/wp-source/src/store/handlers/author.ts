import { Handler } from "../../types";
import { getIdBySlug, populate, getTotal, getTotalPages } from "../helpers";

// CASES:
// 1. !data.isAuthor
// 2. !isPopulating && data.isAuthor && !data.pages[page - 1]

const tagHandler: Handler = async (
  ctx,
  { name, params, page = 1, isPopulating }
) => {
  const state = ctx.state.source;
  const effects = ctx.effects.source;
  
  // 0. Get data from store
  let data = state.data(name);

  // 1. init data if it isn't already
  if (!data.isAuthor) {
    // Search id in state or get it from WP REST API
    const id = await getIdBySlug(ctx, "author", params.slug);

    data = {
      id,
      pages: [],
      isArchive: true,
      isAuthor: true,
      isFetching: true,
    };

    state.dataMap[name] = data;
  }

  // 2. If data is a Tag, then all data is populated
  if (!isPopulating && data.isAuthor && !data.pages[page - 1]) {
    // here we know the id!!
    const { id } = data;
    // and we don't need to init data
    // just get the page we are requesting
    const response = await effects.api.get({
      endpoint: "posts",
      params: { author: id, search: params.s, page, _embed: true }
    });
    // populate response and add page to data
    data.pages[page - 1] = await populate(ctx, response);
    // and assign total and totalPages values
    data.total = getTotal(response);
    data.totalPages = getTotalPages(response);
  }

  // 3. At this point, data is ready to be consumed
  data.isReady = true;
};

export default tagHandler;
