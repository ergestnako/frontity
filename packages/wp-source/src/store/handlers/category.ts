import { Handler } from "../../types";
import {
  getIdBySlug,
  normalize,
  getTotal,
  getTotalPages,
  addPage
} from "./utils";

// 1. category isn't in "source.category"
//    !source.category[catId]
// 2. category exists in "source.category"
//     source.category[catId]
// 3. category exists but not the page (!source.data[name].page[page])
//    !source.data[name].page[page]

const categoryHandler: Handler = async (ctx, { name, params, page = 1 }) => {
  const state = ctx.state.source;
  const actions = ctx.actions.source;
  const effects = ctx.effects.source;

  const catId = await getIdBySlug(ctx, "category", params.slug);

  const data = state.data[name];

  const doesNotExist = !state.category[catId];
  const hasNotPage =
    doesNotExist || !(data.page && data.page[page]);

  let entities: any;
  let total: number;
  let totalPages: number;

  if (doesNotExist || hasNotPage) {
    const response = await effects.api.get({
      endpoint: "posts",
      params: { categories: catId, search: params.s, page }
    });

    // Throw an error if the request has failed
    if (!response.ok) throw new Error();

    entities = await normalize(response);
    total = getTotal(response);
    totalPages = getTotalPages(response);

    // Add entities to the state
    actions.populate({ entities });
  }

  // Init the category if it doesn't exist
  if (doesNotExist) {
    const { link } = state.category[catId];
    Object.assign(data, {
      type: "category",
      id: catId,
      link,
      isArchive: true,
      isTaxonomy: true,
      isCategory: true,
      total,
      totalPages
    });
  }

  // Add the page if it doesn't exist
  if (hasNotPage) addPage(data, page, entities);
};

export default categoryHandler;
