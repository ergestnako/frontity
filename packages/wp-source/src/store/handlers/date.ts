import { Handler } from "../types";

const dateHandler: Handler = async (ctx, { name, params, page = 1 }) => {
  const state = ctx.state.source;
  const actions = ctx.actions.source;
  const effects = ctx.effects.source;

  // Build date property
  const { year, month = "01", day = "01" } = params;
  const date = `${year}-${month}-${day}T00:00:00`;

  const hasNotPage =
    !(state.data[name].page && state.data[name].page[page]);

  let isOk: boolean;
  let entities: any;
  let total: number;
  let totalPages: number;

  if (hasNotPage) {
    ({ isOk, entities, total, totalPages } = await effects.api.get({
      endpoint: "posts",
      params: { after: date, search: params.s, page }
    }));

    // Throw an error if the request has failed
    if (!isOk) throw new Error();

    // Add entities to the state
    actions.populate({ entities });
  }

  // Init the date
  Object.assign(state.data[name], {
    date,
    isArchive: true,
    isDate: true,
    total,
    totalPages
  });

  // Add the page if it doesn't exist
  if (hasNotPage) {
    state.data[name].page = state.data[name].page || [];
    state.data[name].page[page || 1] = entities.map(({ type, id, link }) => ({
      type,
      id,
      link
    }));
  }
};

export default dateHandler;
