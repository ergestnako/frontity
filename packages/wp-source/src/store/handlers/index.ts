import attachmentHandler from "./attachment";
import postOrPageHandler from "./postOrPage";
import authorHandler from "./author";
import categoryHandler from "./category";
import tagHandler from "./tag";
import postArchiveHandler from "./postArchive";
import dateHandler from "./date";

export {
  attachmentHandler,
  postOrPageHandler,
  authorHandler,
  categoryHandler,
  tagHandler,
  postArchiveHandler,
  dateHandler
};
