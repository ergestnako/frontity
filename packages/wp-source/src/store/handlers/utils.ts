import { normalize as norm } from "normalizr";
import * as schemas from "../../schemas";
import { Context } from "../../types";

const typesToEndpoints = {
  author: "users",
  category: "categories",
  tag: "tags",
  post: "posts",
  page: "pages",
  attachment: "media"
};

export const getIdBySlug = async (
  ctx: Context,
  type: string,
  slug: string
): Promise<number> => {
  const state = ctx.state.source;
  const effects = ctx.effects.source;

  // Search an entity with the given slug
  let entity: any = Object.values(state[type]).find(
    (e: any) => e.slug === slug
  );

  // Found: return entity id
  if (entity) return entity.id;

  // Not found: get it from the WP REST API
  const response = await effects.api.get({
    endpoint: typesToEndpoints[type] || "",
    params: { slug }
  });

  [entity] = await response.json();

  return entity && entity.id;
};

// TODO - Promise must not return "any"
export const normalize = async (response: Response): Promise<any> => {
  const json = await response.json();
  // Normalize response
  const { entities } = norm(
    json,
    json instanceof Array ? schemas.list : schemas.entity
  );
  // Return just the attribute 'entities'
  return entities;
};

export const getTotal = (response: Response): number =>
  parseInt(response.headers.get("X-WP-Total"));

export const getTotalPages = (response: Response): number =>
  parseInt(response.headers.get("X-WP-TotalPages"));

export const addPage = (data: any, page: number, entities: any[]) => {
  data.page = data.page || [];
  data.page[page || 1] = entities.map(({ type, id, link }) => ({
    type,
    id,
    link
  }));
};
