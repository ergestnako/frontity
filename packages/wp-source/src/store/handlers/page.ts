import { Handler } from "../types";
import { normalize } from "./utils";

const pageHandler: Handler = async (ctx, { name, params }) => {
  const state = ctx.state.source;
  const actions = ctx.actions.source;
  const effects = ctx.effects.source;

  const { slug } = params;

  let page: any = Object.values(state.page).find((p: any) => p.slug === slug);

  // If not found
  if (!page) {
    const response = await effects.api.get({
      endpoint: "pages",
      params: { slug, _embed: true }
    });

    const entities = await normalize(response);
    [page] = entities;
    actions.populate({ entities });
  }

  // Init data
  const { type, id, link } = page;
  Object.assign(state.data[name], {
    type,
    id,
    link,
    isPostType: true,
    isPage: true
  });
};

export default pageHandler;
