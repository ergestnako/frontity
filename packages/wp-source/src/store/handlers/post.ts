import { Handler } from "../types";
import { normalize } from "./utils";

const postHandler: Handler = async (ctx, { name, params }) => {
  const state = ctx.state.source;
  const actions = ctx.actions.source;
  const effects = ctx.effects.source;

  const { slug } = params;

  let post: any = Object.values(state.post).find((p: any) => p.slug === slug);

  // If not found
  if (!post) {
    const response = await effects.api.get({
      endpoint: "posts",
      params: { slug, _embed: true }
    });

    const entities = await normalize(response);
    [post] = entities;
    actions.populate({ entities });
  }

  // Init data
  const { type, id, link } = post;
  Object.assign(state.data[name], {
    type,
    id,
    link,
    isPostType: true,
    isPost: true,
  });
};

export default postHandler;
