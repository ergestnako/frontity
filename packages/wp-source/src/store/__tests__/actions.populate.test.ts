import { createOvermindMock } from "overmind";
import { namespaced } from "overmind/config";
import * as source from "..";
import { normalize } from "../handlers/utils";
import { Response } from "cross-fetch";

import posts from "./mocks/populate/posts.json";
import pages from "./mocks/populate/pages.json";

describe("actions", () => {
  test.only("populate list of posts", async () => {
    //init store
    const store = createOvermindMock(namespaced({ source }), {
      source: {
        resolver: {
          match: () => ({
            handler: async () => {},
            params: {}
          })
        }
      }
    });

    // add entities to store
    const entities = await normalize(new Response(JSON.stringify(posts)));

    // Check that posts and embedded entities are populated
    const mutations = await store.actions.source.populate({ entities });
    expect(mutations).toMatchSnapshot();
  });

  // test("populate pages", async () => {
  //   // add entities to store
  //   await store.actions.source.populate({ entities: pages });

  //   // check that pages are populated
  //   expect(store.state.source.post).toMatchSnapshot();
  //   expect(store.state.source.category).toMatchSnapshot();
  //   expect(store.state.source.tag).toMatchSnapshot();
  //   expect(store.state.source.attachment).toMatchSnapshot();
  //   expect(store.state.source.author).toMatchSnapshot();
  // });
});
