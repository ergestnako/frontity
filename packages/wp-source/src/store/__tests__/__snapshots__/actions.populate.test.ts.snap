// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions populate list of posts 1`] = `
Array [
  Object {
    "args": Array [
      Object {
        "avatar_urls": Object {
          "24": "http://0.gravatar.com/avatar/?s=24&d=mm&r=g",
          "48": "http://0.gravatar.com/avatar/?s=48&d=mm&r=g",
          "96": "http://0.gravatar.com/avatar/?s=96&d=mm&r=g",
        },
        "description": "",
        "id": 1,
        "link": "http://frontity.test/author/admin/",
        "name": "admin",
        "slug": "admin",
        "type": "author",
        "url": "",
      },
    ],
    "method": "set",
    "path": "source.author.1",
  },
  Object {
    "args": Array [
      Object {
        "isFetching": false,
      },
    ],
    "method": "set",
    "path": "source.data./author/admin/",
  },
  Object {
    "args": Array [
      Object {
        "id": 1,
        "link": "http://frontity.test/category/uncategorized/",
        "name": "Uncategorized",
        "slug": "uncategorized",
        "taxonomy": "category",
        "type": "category",
      },
    ],
    "method": "set",
    "path": "source.category.1",
  },
  Object {
    "args": Array [
      Object {
        "isFetching": false,
      },
    ],
    "method": "set",
    "path": "source.data./category/uncategorized/",
  },
  Object {
    "args": Array [
      Object {
        "id": 4,
        "link": "http://frontity.test/category/aut-vel-dolores-exercitationem-cupiditate-et-quis-aperiam/",
        "name": "Aut vel dolores exercitationem cupiditate et quis aperiam",
        "slug": "aut-vel-dolores-exercitationem-cupiditate-et-quis-aperiam",
        "taxonomy": "category",
        "type": "category",
      },
    ],
    "method": "set",
    "path": "source.category.4",
  },
  Object {
    "args": Array [
      Object {
        "isFetching": false,
      },
    ],
    "method": "set",
    "path": "source.data./category/aut-vel-dolores-exercitationem-cupiditate-et-quis-aperiam/",
  },
  Object {
    "args": Array [
      Object {
        "id": 6,
        "link": "http://frontity.test/category/et-modi-distinctio-voluptatem-quia-cupiditate-labore-corrupti-perferendis-ut-autem/",
        "name": "Et modi distinctio voluptatem quia cupiditate labore corrupti perferendis ut autem",
        "slug": "et-modi-distinctio-voluptatem-quia-cupiditate-labore-corrupti-perferendis-ut-autem",
        "taxonomy": "category",
        "type": "category",
      },
    ],
    "method": "set",
    "path": "source.category.6",
  },
  Object {
    "args": Array [
      Object {
        "isFetching": false,
      },
    ],
    "method": "set",
    "path": "source.data./category/et-modi-distinctio-voluptatem-quia-cupiditate-labore-corrupti-perferendis-ut-autem/",
  },
  Object {
    "args": Array [
      Object {
        "id": 7,
        "link": "http://frontity.test/category/beatae-numquam-tempore-ut-aut-architecto-dolor-adipisci-asperiores-distinctio-facere-perferendis/",
        "name": "Beatae numquam tempore ut aut architecto dolor adipisci asperiores distinctio facere perferendis",
        "slug": "beatae-numquam-tempore-ut-aut-architecto-dolor-adipisci-asperiores-distinctio-facere-perferendis",
        "taxonomy": "category",
        "type": "category",
      },
    ],
    "method": "set",
    "path": "source.category.7",
  },
  Object {
    "args": Array [
      Object {
        "isFetching": false,
      },
    ],
    "method": "set",
    "path": "source.data./category/beatae-numquam-tempore-ut-aut-architecto-dolor-adipisci-asperiores-distinctio-facere-perferendis/",
  },
  Object {
    "args": Array [
      Object {
        "id": 9,
        "link": "http://frontity.test/category/odio-distinctio-odit-numquam-asperiores-neque-eos-sed/",
        "name": "Odio distinctio odit numquam asperiores neque eos sed",
        "slug": "odio-distinctio-odit-numquam-asperiores-neque-eos-sed",
        "taxonomy": "category",
        "type": "category",
      },
    ],
    "method": "set",
    "path": "source.category.9",
  },
  Object {
    "args": Array [
      Object {
        "isFetching": false,
      },
    ],
    "method": "set",
    "path": "source.data./category/odio-distinctio-odit-numquam-asperiores-neque-eos-sed/",
  },
  Object {
    "args": Array [
      Object {
        "id": 13,
        "link": "http://frontity.test/tag/est-velit-earum-ut-nam-voluptates-asperiores-odio-alias-nihil-eum-ratione-et-repudiandae/",
        "name": "Est velit earum ut nam voluptates asperiores odio alias nihil eum ratione et repudiandae",
        "slug": "est-velit-earum-ut-nam-voluptates-asperiores-odio-alias-nihil-eum-ratione-et-repudiandae",
        "taxonomy": "tag",
        "type": "tag",
      },
    ],
    "method": "set",
    "path": "source.tag.13",
  },
  Object {
    "args": Array [
      Object {
        "isFetching": false,
      },
    ],
    "method": "set",
    "path": "source.data./tag/est-velit-earum-ut-nam-voluptates-asperiores-odio-alias-nihil-eum-ratione-et-repudiandae/",
  },
  Object {
    "args": Array [
      Object {
        "id": 15,
        "link": "http://frontity.test/category/dolorem-ut-ea-sit-quidem-et-quia-id-qui-provident/",
        "name": "Dolorem ut ea sit quidem et quia id qui provident",
        "slug": "dolorem-ut-ea-sit-quidem-et-quia-id-qui-provident",
        "taxonomy": "category",
        "type": "category",
      },
    ],
    "method": "set",
    "path": "source.category.15",
  },
  Object {
    "args": Array [
      Object {
        "isFetching": false,
      },
    ],
    "method": "set",
    "path": "source.data./category/dolorem-ut-ea-sit-quidem-et-quia-id-qui-provident/",
  },
  Object {
    "args": Array [
      Object {
        "id": 16,
        "link": "http://frontity.test/category/sit-ut-vero-illo-consequatur-hic-ut-voluptate-qui/",
        "name": "Sit ut vero illo consequatur hic ut voluptate qui",
        "slug": "sit-ut-vero-illo-consequatur-hic-ut-voluptate-qui",
        "taxonomy": "category",
        "type": "category",
      },
    ],
    "method": "set",
    "path": "source.category.16",
  },
  Object {
    "args": Array [
      Object {
        "isFetching": false,
      },
    ],
    "method": "set",
    "path": "source.data./category/sit-ut-vero-illo-consequatur-hic-ut-voluptate-qui/",
  },
  Object {
    "args": Array [
      Object {
        "id": 17,
        "link": "http://frontity.test/tag/a-id-autem-et-esse-id-adipisci/",
        "name": "A id autem et esse id adipisci",
        "slug": "a-id-autem-et-esse-id-adipisci",
        "taxonomy": "tag",
        "type": "tag",
      },
    ],
    "method": "set",
    "path": "source.tag.17",
  },
  Object {
    "args": Array [
      Object {
        "isFetching": false,
      },
    ],
    "method": "set",
    "path": "source.data./tag/a-id-autem-et-esse-id-adipisci/",
  },
  Object {
    "args": Array [
      Object {
        "id": 19,
        "link": "http://frontity.test/tag/possimus-minima-amet-eum-alias-quo-magni-quas-a-est-amet-et-sunt/",
        "name": "Possimus minima amet eum alias quo magni quas a est amet et sunt",
        "slug": "possimus-minima-amet-eum-alias-quo-magni-quas-a-est-amet-et-sunt",
        "taxonomy": "tag",
        "type": "tag",
      },
    ],
    "method": "set",
    "path": "source.tag.19",
  },
  Object {
    "args": Array [
      Object {
        "isFetching": false,
      },
    ],
    "method": "set",
    "path": "source.data./tag/possimus-minima-amet-eum-alias-quo-magni-quas-a-est-amet-et-sunt/",
  },
  Object {
    "args": Array [
      Object {
        "_embedded": Object {
          "author": Array [
            1,
          ],
          "replies": Array [
            Array [
              Object {
                "author": 0,
                "author_avatar_urls": Object {
                  "24": "http://1.gravatar.com/avatar/?s=24&d=mm&r=g",
                  "48": "http://1.gravatar.com/avatar/?s=48&d=mm&r=g",
                  "96": "http://1.gravatar.com/avatar/?s=96&d=mm&r=g",
                },
                "author_name": "A WordPress Commenter",
                "author_url": "https://wordpress.org/",
                "content": Object {
                  "rendered": "<p>Hi, this is a comment.<br />
To get started with moderating, editing, and deleting comments, please visit the Comments screen in the dashboard.<br />
Commenter avatars come from <a href=\\"https://gravatar.com\\">Gravatar</a>.</p>
",
                },
                "date": "2019-03-11T11:46:53",
                "id": 1,
                "link": "http://frontity.test/hello-world/#comment-1",
                "parent": 0,
                "type": "comment",
              },
            ],
          ],
          "wp:term": Array [
            Array [
              1,
            ],
            Array [],
            Array [],
          ],
        },
        "author": 1,
        "categories": Array [
          1,
        ],
        "comment_status": "open",
        "content": Object {
          "protected": false,
          "rendered": "
<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>
",
        },
        "date": "2019-03-11T11:46:53",
        "date_gmt": "2019-03-11T11:46:53",
        "excerpt": Object {
          "protected": false,
          "rendered": "<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>
",
        },
        "featured_media": 0,
        "format": "standard",
        "guid": Object {
          "rendered": "http://frontity.test/?p=1",
        },
        "id": 1,
        "link": "http://frontity.test/hello-world/",
        "meta": Array [],
        "modified": "2019-03-11T11:46:53",
        "modified_gmt": "2019-03-11T11:46:53",
        "news": Array [],
        "ping_status": "open",
        "slug": "hello-world",
        "status": "publish",
        "sticky": false,
        "tags": Array [],
        "taxonomiesMap": Object {
          "category": Array [
            1,
          ],
        },
        "template": "",
        "title": Object {
          "rendered": "Hello world!",
        },
        "type": "post",
      },
    ],
    "method": "set",
    "path": "source.post.1",
  },
  Object {
    "args": Array [
      Object {
        "isFetching": false,
      },
    ],
    "method": "set",
    "path": "source.data./hello-world/",
  },
  Object {
    "args": Array [
      Object {
        "_embedded": Object {
          "author": Array [
            1,
          ],
          "wp:featuredmedia": Array [
            84,
          ],
          "wp:term": Array [
            Array [
              7,
              16,
              1,
            ],
            Array [
              17,
              13,
              19,
            ],
            Array [],
          ],
        },
        "author": 1,
        "categories": Array [
          7,
          16,
          1,
        ],
        "comment_status": "open",
        "content": Object {
          "protected": false,
          "rendered": "<blockquote><p>Odio tenetur qui veniam. Facere eaque voluptate enim facilis. Nemo omnis sed rerum velit.</p></blockquote>
",
        },
        "date": "2019-02-13T15:37:00",
        "date_gmt": "2019-02-13T15:37:00",
        "excerpt": Object {
          "protected": false,
          "rendered": "<p>Odio tenetur qui veniam. Facere eaque voluptate enim facilis. Nemo omnis sed rerum velit.</p>
",
        },
        "featured_media": 84,
        "format": "standard",
        "guid": Object {
          "rendered": "http://frontity.test/veniam-reiciendis-esse-aliquid/",
        },
        "id": 83,
        "link": "http://frontity.test/veniam-reiciendis-esse-aliquid/",
        "meta": Array [],
        "modified": "2019-02-13T15:37:00",
        "modified_gmt": "2019-02-13T15:37:00",
        "news": Array [],
        "ping_status": "open",
        "slug": "veniam-reiciendis-esse-aliquid",
        "status": "publish",
        "sticky": false,
        "tags": Array [
          17,
          13,
          19,
        ],
        "taxonomiesMap": Object {
          "category": Array [
            7,
            16,
            1,
          ],
          "tag": Array [
            17,
            13,
            19,
          ],
        },
        "template": "",
        "title": Object {
          "rendered": "Veniam reiciendis esse aliquid",
        },
        "type": "post",
      },
    ],
    "method": "set",
    "path": "source.post.83",
  },
  Object {
    "args": Array [
      Object {
        "isFetching": false,
      },
    ],
    "method": "set",
    "path": "source.data./veniam-reiciendis-esse-aliquid/",
  },
  Object {
    "args": Array [
      Object {
        "_embedded": Object {
          "author": Array [
            1,
          ],
          "wp:featuredmedia": Array [
            113,
          ],
          "wp:term": Array [
            Array [
              4,
              15,
              6,
              9,
            ],
            Array [
              17,
              13,
              19,
            ],
            Array [],
          ],
        },
        "author": 1,
        "categories": Array [
          4,
          15,
          6,
          9,
        ],
        "comment_status": "open",
        "content": Object {
          "protected": false,
          "rendered": "<h1>Mollitia debitis commodi dolor suscipit voluptas voluptatem dolor. Voluptate eligendi suscipit voluptates in. Velit perferendis saepe consequuntur rem animi</h1>
",
        },
        "date": "2019-01-09T13:14:41",
        "date_gmt": "2019-01-09T13:14:41",
        "excerpt": Object {
          "protected": false,
          "rendered": "<p>Mollitia debitis commodi dolor suscipit voluptas voluptatem dolor. Voluptate eligendi suscipit voluptates in. Velit perferendis saepe consequuntur rem animi</p>
",
        },
        "featured_media": 113,
        "format": "standard",
        "guid": Object {
          "rendered": "http://frontity.test/eum-ipsum-eos-sit-sed-sint/",
        },
        "id": 112,
        "link": "http://frontity.test/eum-ipsum-eos-sit-sed-sint/",
        "meta": Array [],
        "modified": "2019-01-09T13:14:41",
        "modified_gmt": "2019-01-09T13:14:41",
        "news": Array [],
        "ping_status": "closed",
        "slug": "eum-ipsum-eos-sit-sed-sint",
        "status": "publish",
        "sticky": false,
        "tags": Array [
          17,
          13,
          19,
        ],
        "taxonomiesMap": Object {
          "category": Array [
            4,
            15,
            6,
            9,
          ],
          "tag": Array [
            17,
            13,
            19,
          ],
        },
        "template": "",
        "title": Object {
          "rendered": "Eum ipsum eos sit sed sint",
        },
        "type": "post",
      },
    ],
    "method": "set",
    "path": "source.post.112",
  },
  Object {
    "args": Array [
      Object {
        "isFetching": false,
      },
    ],
    "method": "set",
    "path": "source.data./eum-ipsum-eos-sit-sed-sint/",
  },
  Object {
    "args": Array [
      Object {
        "alt_text": "",
        "author": 1,
        "caption": Object {
          "rendered": "",
        },
        "date": "2019-04-10T09:47:02",
        "id": 84,
        "link": "http://frontity.test/84/",
        "media_details": Object {},
        "media_type": "image",
        "mime_type": "image/jpeg",
        "slug": "84",
        "source_url": "http://frontity.test/wp-content/uploads/2019/04/f561cebe-f042-37cc-ba31-cf202fb98a1b.jpg",
        "title": Object {
          "rendered": "",
        },
        "type": "attachment",
      },
    ],
    "method": "set",
    "path": "source.attachment.84",
  },
  Object {
    "args": Array [
      Object {
        "isFetching": false,
      },
    ],
    "method": "set",
    "path": "source.data./84/",
  },
  Object {
    "args": Array [
      Object {
        "alt_text": "",
        "author": 1,
        "caption": Object {
          "rendered": "",
        },
        "date": "2019-04-10T09:47:34",
        "id": 113,
        "link": "http://frontity.test/113/",
        "media_details": Object {},
        "media_type": "image",
        "mime_type": "image/jpeg",
        "slug": "113",
        "source_url": "http://frontity.test/wp-content/uploads/2019/04/9ee08884-e986-37bb-9e3e-6c742af8ff9c.jpg",
        "title": Object {
          "rendered": "",
        },
        "type": "attachment",
      },
    ],
    "method": "set",
    "path": "source.attachment.113",
  },
  Object {
    "args": Array [
      Object {
        "isFetching": false,
      },
    ],
    "method": "set",
    "path": "source.data./113/",
  },
]
`;
